preset/kaleidoscope/kaleidoscope.cc

```diff
@@ -153,6 +148,47 @@ void Kaleidoscope::OnDrawFrame(absl::Span<const float> samples,
   glViewport(0, 0, width(), height());
   rectangle.Draw();
 
+  glFlush();
+
+  {
+    auto back_activation = back_render_target_->Activate();
+
+    warp_program_->Use();
+
+    LOG(DEBUG) << "Using program";
```

Comment:
Since this code appears in two places, could you factor it out into a helper function in the anonymous namespace at the top of this file?